lib/snippets/release-gem

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env ruby
+
+require 'ostruct'
+require 'net/https'
+require 'uri'
+
+class FakeSpec < OpenStruct
+  class << self
+    def load(gemspec_path)
+      fake_spec = new
+      original_spec_constructor = ::Gem::Specification.method(:new)
+      ::Gem::Specification.define_singleton_method(:new) { |&block| block.call(fake_spec) }
+      Kernel.load(gemspec_path)
+      ::Gem::Specification.singleton_class.send(:remove_method, :new)
+      ::Gem::Specification.define_singleton_method(:new, original_spec_constructor)
+      fake_spec
+    end
+  end
+
+  private
+
+  def method_missing(name, *args)
+    super
+  rescue NoMethodError
+  end
+end
+
+module RubygemsAPI
+  extend self
+
+  def published?(name, version)
+    uri = URI.parse("https://rubygems.org/api/v2/rubygems/#{name}/versions/#{version}.json")
+    Net::HTTP.get_response(uri).is_a?(Net::HTTPSuccess)
+  end
+end
+
+spec_path, *release_command = ARGV
+release_command = %w(bundle exec rake release) if release_command.empty?
+
+spec = FakeSpec.load(spec_path)
```

Comment:
Rather than parsing the gemspec, would it make sense to `gem build` and inspect the resulting `.gem`?

Reply:
`Gem::Package.new('pkg/mygem-2.2.0.gem').spec.version` will read the version from the compiled spec.

Reply:
I'd rather not, because it's not guaranteed a gem will properly build with `gem build`, it might require previous step, like `ci-queue` does: https://github.com/Shopify/ci-queue/blob/0a884b216199e12f769292c17b2ad7ab16186c6a/ruby/Rakefile#L26

However that whole FakeSpec is a bit over the top, I cargo culted it from the previous script, but can't remember a justification for it.

I replaced it by a simple `Gem::Specification.load`. SHould be much cleaner now.